caddy/directives.go

```diff
@@ -68,6 +68,23 @@ var directiveOrder = []directive{
 	{"browse", setup.Browse},
 }
 
+// RegisterDirective adds the given directive to caddy's list of directives.
+// Pass the name of a directive you want it to be placed after,
+// otherwise it will be placed at the bottom of the stack.
+func RegisterDirective(name string, setup SetupFunc, after string) {
+	dir := directive{name: name, setup: setup}
+	idx := len(directiveOrder)
+	for i := range directiveOrder {
+		if directiveOrder[i].name == after {
+			idx = i + 1
+			break
+		}
+	}
```

Comment:
Actually I've been thinking about this... would it make more sense to change "after" to "before"? That way, if it's empty (default value), it should go at the end of the stack, which is common for simple testing situations (like with caddydev).